packages/typescript-estree/src/node-utils.ts

```diff
@@ -766,3 +766,18 @@ export function getContainingFunction(
 ): ts.SignatureDeclaration | undefined {
   return ts.findAncestor(node.parent, ts.isFunctionLike);
 }
+
+// `ts.nodeCanBeDecorated`
+export function nodeCanBeDecorated(node: ts.Node): boolean {
+  return [true, false].some(
+    useLegacyDecorators =>
+      // @ts-expect-error -- internal api
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      ts.nodeCanBeDecorated?.(
```

Comment:
we can't use this function, or at least not without version checks.
in versions before 5.0 this function did not take `boolean` as the first argument - so by calling it like this we will cause a crash.

Given it's an internal API - I think we might want to consider rewriting it for ourselves? The logic will be pretty static given the spec doesn't change much if at all.

Reply:
`hasAmbientModifier` was only added in TS 4.3, and currently on v6 we are supporting 4.2
However based on the 2y lifecycle of versions that DefinitelyTyped supports (which we want to roughly align with) we will actually now be able to bump the minimum version to 4.3.

We can queue a separate PR for that then update this PR to use `ts.hasAmbientModifier`

Reply:
`ts.hasAmbientModifier` seems not available https://github.com/typescript-eslint/typescript-eslint/actions/runs/4739172024/jobs/8413765640?pr=6723#step:4:98

Reply:
it's still a private, internal API!

Reply:
Let's use a fake one for now?  https://github.com/typescript-eslint/typescript-eslint/pull/6723/commits/a84fa78fd3fa020ba32ae221025058ef0c128d11

Reply:
If we're going with a fake `() => false` one then I just removed it altogether. Seems like everything is working well without it? I posted a link in https://github.com/microsoft/TypeScript/issues/52727#issuecomment-1595567869.